### PR TITLE
fix(core/pubsubhub): account for all message types

### PR DESCRIPTION
**Pull request by marcoscaceres**
Events were being broadcast to no one... this now forces every registered `pub()` to have a listener of a particular name.

Hopefully we can get rid of the eventing model in the future. It creates too many interdependent parts.

### CODE REVIEW

File: tests/spec/core/validators-spec.js [hunk 1]
```diff
@@ -30,6 +31,13 @@ describe("Core - Validators", () => {
     it("generates an error if the element name is not valid", () => {
       const elements = ["my element", "crypto$", "🪳", "-something", ""];
       for (const element of elements) {
+        sub(
```
Comment (sidvishnoi) [inline tests/spec/core/validators-spec.js:34]: We should probably use a spy here, and/or use jasmine's `toHaveBeenCalledWith`

File: src/core/pubsubhub.js [hunk 2]
```diff
@@ -12,9 +12,14 @@ export const name = "core/pubsubhub";
 
 const subscriptions = new Map();
 
+/**
+ *
+ * @param {EventTopic} topic
+ * @param  {...any} data
```
Comment (sidvishnoi) [inline src/core/pubsubhub.js:18]: Optional TypeScript improvement: We can make EventTopic an interface and make these functions generic over that. Then we would get proper type checking in all pubsub calls.

File: src/core/examples.js [hunk 3]
```diff
@@ -113,7 +111,6 @@ export function run() {
       const selfLink = div.querySelector("a.self-link");
       selfLink.href = `#${div.id}`;
       example.replaceWith(div);
-      if (!inAside) pub("example", report);
```
Comment (sidvishnoi) [inline src/core/examples.js:116]: Nice! _Begone!_

File: src/core/pubsubhub.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/pubsubhub.js:18]: Yeah, was thinking the same... but I need to refactor all the call sites... they only use a single argument.

I found a huge bug also with pubsubhub.js... it wasn't designed to handle async functions, and now we have a mess for supporting "conf.beforeSave".

File: tests/spec/core/validators-spec.js (hunk 1 above)
Comment (marcoscaceres) [reply tests/spec/core/validators-spec.js:34]: Nearly went there, but this seemed cleaner to me... the spy adds a lot of indirection.

Comment (marcoscaceres) [reply tests/spec/core/validators-spec.js:34]: Got rid of the actual test part, as it's not important for these.

Comment (marcoscaceres) [reply tests/spec/core/validators-spec.js]: Plus a lot of these tests still check "respec-offending-element" :( will need to rewrite them later.

File: tests/unit/SpecHelper.js [hunk 4]
```diff
@@ -14,6 +14,11 @@ export function makePluginDoc(
   plugins,
   { config = {}, head = `<meta charset="UTF-8" />`, body = "" } = {}
 ) {
+  plugins = [
+    `/src/core/ui.js`, // Needed for "start-all" event
+    `/src/core/dfn.js`, // Needed for "plugins-done" event,
```
Comment (sidvishnoi) [inline tests/unit/SpecHelper.js:19]: Nit: Let's either move this to `allPlugins` line below or move the `/src/core/base-runner.js` here,

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**